# HR Analytics: Workforce Performance & Employee Insights

## 02 - Data Quality Assessment

### Project Overview

Data quality assessment is a critical step before performing exploratory analysis. This notebook evaluates the completeness, consistency, and reliability of the dataset by identifying missing values, duplicate records, incorrect data types, and potential data quality issues.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)
pd.set_option("display.float_format", "{:,.2f}".format)

df = pd.read_csv("../data/HRDataset_v14.csv")

#### Dataset Shape

Display the number of rows and columns contained in the dataset.

In [2]:
df.shape

(311, 36)

#### Missing Values Analysis

Identify the number of missing values for every variable.

In [3]:
missing_values = df.isnull().sum().sort_values(ascending=False)

missing_values

DateofTermination             207
ManagerID                       8
EmpID                           0
RaceDesc                        0
DateofHire                      0
TermReason                      0
EmploymentStatus                0
Department                      0
ManagerName                     0
Employee_Name                   0
RecruitmentSource               0
PerformanceScore                0
EngagementSurvey                0
EmpSatisfaction                 0
SpecialProjectsCount            0
LastPerformanceReview_Date      0
DaysLateLast30                  0
HispanicLatino                  0
CitizenDesc                     0
MaritalDesc                     0
FromDiversityJobFairID          0
MarriedID                       0
MaritalStatusID                 0
GenderID                        0
EmpStatusID                     0
DeptID                          0
PerfScoreID                     0
Salary                          0
Sex                             0
Termd         

#### Missing Values Percentage

Calculate the percentage of missing values for each feature.

In [4]:
missing_percentage = (
    df.isnull().sum() / len(df) * 100
).sort_values(ascending=False)

missing_percentage

DateofTermination            66.56
ManagerID                     2.57
EmpID                         0.00
RaceDesc                      0.00
DateofHire                    0.00
TermReason                    0.00
EmploymentStatus              0.00
Department                    0.00
ManagerName                   0.00
Employee_Name                 0.00
RecruitmentSource             0.00
PerformanceScore              0.00
EngagementSurvey              0.00
EmpSatisfaction               0.00
SpecialProjectsCount          0.00
LastPerformanceReview_Date    0.00
DaysLateLast30                0.00
HispanicLatino                0.00
CitizenDesc                   0.00
MaritalDesc                   0.00
FromDiversityJobFairID        0.00
MarriedID                     0.00
MaritalStatusID               0.00
GenderID                      0.00
EmpStatusID                   0.00
DeptID                        0.00
PerfScoreID                   0.00
Salary                        0.00
Sex                 

#### Duplicate Records

Check whether duplicate observations exist in the dataset.

In [5]:
df.duplicated().sum()

np.int64(0)

#### Data Types Review

Review the data type of every feature.

In [6]:
df.dtypes

Employee_Name                  object
EmpID                           int64
MarriedID                       int64
MaritalStatusID                 int64
GenderID                        int64
EmpStatusID                     int64
DeptID                          int64
PerfScoreID                     int64
FromDiversityJobFairID          int64
Salary                          int64
Termd                           int64
PositionID                      int64
Position                       object
State                          object
Zip                             int64
DOB                            object
Sex                            object
MaritalDesc                    object
CitizenDesc                    object
HispanicLatino                 object
RaceDesc                       object
DateofHire                     object
DateofTermination              object
TermReason                     object
EmploymentStatus               object
Department                     object
ManagerName 

#### Unique Values

Determine the number of unique values for every column.

In [7]:
df.nunique().sort_values()

Termd                           2
MarriedID                       2
Sex                             2
GenderID                        2
FromDiversityJobFairID          2
CitizenDesc                     3
EmploymentStatus                3
PerformanceScore                4
PerfScoreID                     4
HispanicLatino                  4
MaritalDesc                     5
EmpStatusID                     5
MaritalStatusID                 5
EmpSatisfaction                 5
DeptID                          6
Department                      6
RaceDesc                        6
DaysLateLast30                  7
RecruitmentSource               9
SpecialProjectsCount            9
TermReason                     18
Absences                       20
ManagerName                    21
ManagerID                      23
State                          28
PositionID                     30
Position                       32
DateofTermination              96
DateofHire                    101
EngagementSurv

#### Numerical Summary

Generate descriptive statistics for numerical variables.

In [8]:
df.describe()

,EmpID,MarriedID,MaritalStatusID,GenderID,EmpStatusID,DeptID,PerfScoreID,FromDiversityJobFairID,Salary,Termd,PositionID,Zip,ManagerID,EngagementSurvey,EmpSatisfaction,SpecialProjectsCount,DaysLateLast30,Absences
count,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,311.00,303.00,311.00,311.00,311.00,311.00,311.00
mean,"10,156.00",0.40,0.81,0.43,2.39,4.61,2.98,0.09,"69,020.68",0.33,16.85,"6,555.48",14.57,4.11,3.89,1.22,0.41,10.24
std,89.92,0.49,0.94,0.50,1.79,1.08,0.59,0.29,"25,156.64",0.47,6.22,"16,908.40",8.08,0.79,0.91,2.35,1.29,5.85
min,"10,001.00",0.00,0.00,0.00,1.00,1.00,1.00,0.00,"45,046.00",0.00,1.00,"1,013.00",1.00,1.12,1.00,0.00,0.00,1.00
25%,"10,078.50",0.00,0.00,0.00,1.00,5.00,3.00,0.00,"55,501.50",0.00,18.00,"1,901.50",10.00,3.69,3.00,0.00,0.00,5.00
50%,"10,156.00",0.00,1.00,0.00,1.00,5.00,3.00,0.00,"62,810.00",0.00,19.00,"2,132.00",15.00,4.28,4.00,0.00,0.00,10.00
75%,"10,233.50",1.00,1.00,1.00,5.00,5.00,3.00,0.00,"72,036.00",1.00,20.00,"2,355.00",19.00,4.70,5.00,0.00,0.00,15.00
max,"10,311.00",1.00,4.00,1.00,5.00,6.00,4.00,1.00,"250,000.00",1.00,30.00,"98,052.00",39.00,5.00,5.00,8.00,6.00,20.00


#### Categorical Summary

Generate descriptive statistics for categorical variables.

In [9]:
df.describe(include="object")

,Employee_Name,Position,State,DOB,Sex,MaritalDesc,CitizenDesc,HispanicLatino,RaceDesc,DateofHire,DateofTermination,TermReason,EmploymentStatus,Department,ManagerName,RecruitmentSource,PerformanceScore,LastPerformanceReview_Date
count,311,311,311,311,311,311,311,311,311,311,104,311,311,311,311,311,311,311
unique,311,32,28,307,2,5,3,4,6,101,96,18,3,6,21,9,4,137
top,"Adinolfi, Wilson K",Production Technician I,MA,09/09/65,F,Single,US Citizen,No,White,1/10/2011,8/19/2018,N/A-StillEmployed,Active,Production,Michael Albert,Indeed,Fully Meets,1/14/2019
freq,1,137,276,2,176,137,295,282,187,14,2,207,207,209,22,87,243,18


#### Data Quality Conclusion

This assessment evaluates:

- Missing values
- Duplicate records
- Data types
- Unique values
- Overall dataset consistency

The findings from this notebook will guide the data preparation and exploratory analysis performed in the next stage of the project.

#### Final Assessment

The dataset demonstrates a high level of quality suitable for business analytics.

##### Key findings

- No duplicate observations were identified.
- Most variables contain complete information.
- The missing values in `DateofTermination` are expected because active employees have not been terminated.
- The small number of missing values in `ManagerID` represents only 2.57% of the dataset and does not materially impact the analysis.

Overall, the dataset is considered reliable for exploratory analysis, feature engineering, SQL business analysis, and dashboard development.